### DISCUSSION TITLE
Securing endpoints

### DISCUSSION THREAD

**Discussion by glowkeeper** [directus/directus#4551] [Archive]
Is it possible to make certain custom endpoints available only to certain users (or roles)? Maybe via some _service_?

**Comment by rijkvanzanten** [comment]
The current user info is accessible in `req.accountability`:

```js
module.exports = function registerEndpoint(router) {
	router.get('/', (req, res) => {
		if (req.accountability?.admin !== true) return res.send('Nah fam');

		res.send('Hello, Admin!'));
	}
};
```

This object contains the following properties:

https://github.com/directus/directus/blob/40f53fdeb0e702b93d4dce204792c6e66145eaa5/api/src/types/accountability.ts#L1-L9

**Reply by glowkeeper** [reply]
Thanks!

**Reply by glowkeeper** [reply]
Eek! I can't get this to work :( How do users pass their info to the custom endpoint? I tried calling it by setting **Authorization: `Bearer ${access_token}`** in the header (where `access_token` is a logged in _user_), but my endpoint just sees this:
```
Accountability:  {
  user: null,
  role: null,
  admin: false,
  ip: '127.0.0.1',
  userAgent: 'Mozilla/5.0 (Macintosh; Intel Mac OS X 11_2_2) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/89.0.4389.82 Safari/537.36'
}
```

**Reply by keesvanbemmel** [reply]
Did you get this resolve glowkeeper? I'm running into the same challenge

**Reply by keesvanbemmel** [reply]
Just fixed it by using the access_token in the url. Not the best solution, but works for now.

**Reply by glowkeeper** [reply]
Yeah - got it working - see here:  https://github.com/directus/directus/discussions/4597
